persist/fs/seek.go

```diff
@@ -320,11 +340,7 @@ func (s *seeker) SeekByID(id ts.ID) (checked.Bytes, error) {
 // instead of looking it up on its own. Useful in cases where you've already
 // obtained an entry and don't want to waste resources looking it up again.
 func (s *seeker) SeekByIndexEntry(entry IndexEntry) (checked.Bytes, error) {
-	_, err := s.dataFd.Seek(entry.Offset, 0)
-	if err != nil {
-		return nil, err
-	}
-	s.dataReader.Reset(s.dataFd)
+	s.dataReader.Reset(bytes.NewReader(s.dataMmap[entry.Offset:]))
```

Comment:
We should definitely cache the `*bytes.Reader`, perhaps create one in the seeker ctor?

Reply:
As discussed, we can go without any readers for the data mmap thankfully (yay)